src/DynamoCore/Interfaces/IDynamoModel.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Reflection;
+
+namespace Dynamo.Interfaces
+{
+    public interface IDynamoModel : IDisposable
+    {
+        ICustomNodeManager CustomNodeManager { get; }
+        void LoadNodeLibraryFromAssembly(Assembly library);
+    }
```

Comment:
I'd like to do this, but it conflicts with the way that `IExtension` is implemented today. `IExtension` is implemented so that you can access certain parts of `IDynamoModel` at certain points in it's lifetime. This gives the client (i.e. someone implementing an `IExtension`) certain guarantees on whether an object is loaded or not.